subprojects/internal-integ-testing/src/test/groovy/org/gradle/integtests/fixtures/logging/GroupedOutputFixtureTest.groovy

```diff
@@ -327,4 +327,38 @@ Bye world
         then:
         groupedOutput.task(':buildSrc:helloWorld').output == 'Hello world'
     }
+
+    def "striped output remove all ANSI control sequences and work-in-progress area"() {
+        def consoleOutput = """
+\u001B[1m> Task :buildSrc:helloWorld\u001B[m
+Hello world
+
+
+\u001B[2A\u001B[1m<-------------> 0% CONFIGURING [0s]\u001B[m\u001B[35D\u001B[1B\u001B[1m> root project > Compiling /Users/daniel/gradle/gradle/build/tmp/test files/ConsoleBuildSrcFunctionalTest/can_group_task_outp..._buildSrc/pco71/build.gradle into local compilation cache\u001B[m\u001B[185D\u001B[1B\u001B[2A\u001B[0K
+\u001B[1m> Task :byeWorld\u001B[m\u001B[0K
+Bye world
+
+
+\u001B[32;1mBUILD SUCCESSFUL\u001B[0;39m in 0s
+1 actionable task: 1 executed
+\u001B[2K"""
+
+        when:
+        GroupedOutputFixture groupedOutput = new GroupedOutputFixture(consoleOutput)
+
+        then:
+        groupedOutput.stripedOutput == '''
```

Comment:
Some test depends on the full ANSI output. We would need to revisit this to better incorporate the rich and plain output grepping from the test.